### DISCUSSION TITLE
IDEA: Package Scene WITH Scripts

### DISCUSSION THREAD

**Discussion by shak2** [godotengine/godot-proposals#9973] [General Discussions]
Users should be able to package scenes with scripts.

Right now if i'm sharing my scenes I have to include the scripts separately, AND tell people how to structure the thing and where to add the scripts. then they have to do all that and not make mistakes.

Encapsulation is a stated goal of Godot.

Is packaging scenes WITH scripts possible? Can this be added? 4.3?

Also, how about a third type of package scene of  nodes + scripts + resources for 5.0?

**Comment by AThousandShips** [comment]
Have you tried using built in scripts and bundling the resources into the scene?

I'd say though that just putting the scenes and required resources and scripts in the same folder should be enough for anyone to follow instructions on how to use, and is the [suggested way](https://docs.godotengine.org/en/latest/tutorials/best_practices/project_organization.html#organization) to organized your projects

**Comment by theraot** [comment]
I think fix broken dependencies should be working (once more). I'm not saying it is ideal, but if you send that scene with scripts, and they use fix dependencies on their end, I'd expect it to work.
